test/apps/iodemo/io_demo.cc

```diff
@@ -217,6 +227,54 @@ class P2pDemoCommon : public UcxContext {
         return conn->send_io_message(m->buffer(), opts().iomsg_size, m);
     }
 
+    bool send_recv_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                send_recv_data_t send_recv_data, UcxCallback* callback = EmptyCallback::get()) {
+        size_t left_size = data_size;
+        while (left_size > 0) {
+            size_t sz = (left_size >= opts().chunk_size) ? opts().chunk_size : left_size;
+            if (send_recv_data == SEND_DATA) {
+                conn->send_data(buffer(data_size - left_size), sz, sn, callback);
+            } else {
+                conn->recv_data(buffer(data_size - left_size), sz, sn, callback);
+            }
+            left_size -= sz;
+        }
+        return true;
+    }
+
+    bool send_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                                     UcxCallback* callback = EmptyCallback::get()) {
+        return send_recv_data_as_chunks(conn, data_size, sn, SEND_DATA, callback);
+    }
+
+    bool recv_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                                     UcxCallback* callback = EmptyCallback::get()) {
+        return send_recv_data_as_chunks(conn, data_size, sn, RECV_DATA, callback);
+    }
+
+    uint32_t get_chunk_cnt(size_t data_size) {
+        return (opts().chunk_size == 0) ? 1 : ((data_size + opts().chunk_size - 1)
+                                               / opts().chunk_size);
+    }
+    
+    bool send_data(UcxConnection* conn, size_t data_size, uint32_t sn,
+                       UcxCallback* callback = EmptyCallback::get()) {
+        if (opts().chunk_size == 0) {
```

Comment:
default chunk size can be SIZE_MAX, which will always send one chunk. no need special case for chunks on/off

Reply:
@yosefe I also considered as this. But SIZE_MAX/data_size always can be set/changed at run time,  however the default value of chunk size is set at the very beginning. It's inconvient to change the value of chunk size according to SIZE_MAX/data_size at run time. Prefer to make chunk size fixed during run time. Let me know please if you still don't agree this.

Reply:
i don't agree: whenever you use "0" for chunk_size, just replace it by SIZE_MAX...

Reply:
@yosefe What does SIZE_MAX indicate? data_size? We have several different data_size values in io demo, such as hdr->data_size, and the data_size values got get_data_size() with random method. So for different send/receive the data size can be different.

Reply:
SIZE_MAX would be the maximal chunk size to send. so data_size will always be smaller than CHUNK_SIZE in this case

Reply:
@yosefe So what value should it be? Maybe set it the same as test_opts->max_data_size?

Reply:
Fixed. Thanks!